cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -577,12 +579,12 @@ private String subQuery(String index,CQLTermNode node, Table correlation) throws
         termString = String.format(template, "'" + Cql2SqlUtil.cql2string(term) + "'");
         indexString = String.format(template, whereField);
       }
-      selectString = "Cast ( " + targetField + "as UUID)";
+      selectString =  targetField ;
       inKeyword = myField + " IN ";
       whereClause = " WHERE " + indexString + " = " + termString;
     } else {
       inKeyword = myField + " IN ";
-      selectString = "Cast ( " + index2sqlText(foreignTableJsonb, foreignTarget[1]) + "as UUID)";
+      selectString = indexInTable ?  PK_COLUMN_NAME : "Cast ( " + index2sqlText(foreignTableJsonb,  foreignTarget[1]) + " as UUID)";
     }
 
     return inKeyword + " ( SELECT " + selectString + " from " + foreignTarget[0] + whereClause + ")";
```

Comment:
The 9 string variables make it hard to follow how the SQL string is constructed and what the SQL query makes.

Can you at least extract the most easy case, for example like this?

    
    if (tableNamePattern.matcher(term).matches()) {
      if (indexInTable) {
        return myField + " IN ( SELECT id FROM " + foreignTarget[0] + ")";
      }
      return myField + " IN ( SELECT Cast ( " + index2sqlText(foreignTableJsonb,  foreignTarget[1]) + " as UUID)" 
            + " FROM " + foreignTarget[0] + ")";
    }

Reply:
I removed a lot of duplication in this code block.  I am not sure this is needed anymore.